src/resmom/mom_pmix.c

```diff
@@ -0,0 +1,1537 @@
+ * @param[in] ninfo - number of info array entries
+ * @param[in] cbfunc - optional callback function
+ * @param[in] cbdata - opaque data provided to cbfunc
+ *
+ * @return pmix_status_t
+ * @retval PMIX_SUCCESS - request in progress, cbfunc should not be called
+ *                        here but will be called later by PMIx
+ * @retval PMIX_OPERATION_SUCCEEDED - request immediately processed and
+ *                                    successful, cbfunc will not be called
+ * @retval PMIX_ERR_BAD_PARAM - one of the provided parameters was invalid
+ * @retval PMIX_ERR_NOT_IMPLEMENTED - function not implemented
+ */
+static pmix_status_t
+pbs_pmix_deregister_events(
+	pmix_status_t *codes,
+	size_t ncodes,
+	pmix_op_cbfunc_t cbfunc,
+	void *cbdata)
+{
+	log_event(PBSEVENT_DEBUG3, 0, LOG_DEBUG, __func__, "called");
+	log_event(PBSEVENT_DEBUG3, 0, LOG_DEBUG, __func__, "returning");
+	return PMIX_ERR_NOT_IMPLEMENTED;
+}
+
+/**
+ * @brief
+ * Initialize the PMIx server
+ *
+ * @param[in] name - name of daemon (used for logging)
+ *
+ * @return void
+ *
+ * @note
+ * The PMIx library spawns threads from pbs_mom to act as the PMIx server
+ * for applications (PMIx clients) assigned to this vnode. The pbs_mom acts
+ * as the PMIx server even though all it does is call PMIx library functions.
+ * It also means that if pbs_mom exits, any PMIx clients (PMIx enabled
+ * applications) will lose their local server and fail.
+ */
+void
+pbs_pmix_server_init(char *name)
+{
+	pmix_status_t pstat;
+	pmix_server_module_t pbs_pmix_server_module = {
+		/* v1x interfaces */
+		.client_connected = pbs_pmix_client_connected,
+		.client_finalized = pbs_pmix_client_finalized,
+		.abort = pbs_pmix_abort,
+		.fence_nb = pbs_pmix_fence_nb,
+		.direct_modex = pbs_pmix_direct_modex,
+		.publish = pbs_pmix_publish,
+		.lookup = pbs_pmix_lookup,
+		.unpublish = pbs_pmix_unpublish,
+		.spawn = pbs_pmix_spawn,
+		.connect = pbs_pmix_connect,
+		.disconnect = pbs_pmix_disconnect,
+		.register_events = pbs_pmix_register_events,
+		.deregister_events = pbs_pmix_deregister_events,
+		.listener = NULL
+#if PMIX_VERSION_MAJOR > 1
+		,
+		/* v2x interfaces */
+		.notify_event = NULL,
+		.query = NULL,
+		.tool_connected = NULL,
+		.log = NULL,
+		.allocate = NULL,
+		.job_control = NULL,
+		.monitor = NULL
+#endif
+#if PMIX_VERSION_MAJOR > 2
+		,
+		/* v3x interfaces */
+		.get_credential = NULL,
+		.validate_credential = NULL,
+		.iof_pull = NULL,
+		.push_stdin = NULL
+#endif
+#if PMIX_VERSION_MAJOR > 3
+		,
+		/* v4x interfaces */
+		.group = NULL
+#endif
+	};
+
+	pstat = PMIx_server_init(&pbs_pmix_server_module, NULL, 0);
+	if (pstat != PMIX_SUCCESS) {
+		snprintf(log_buffer, sizeof(log_buffer),
+			"Could not initialize PMIx server: %s",
+			PMIx_Error_string(pstat));
+		log_event(PBSEVENT_ERROR, PBS_EVENTCLASS_SERVER,
+			LOG_ERR, name, log_buffer);
+	} else {
+		log_event(PBSEVENT_DEBUG, PBS_EVENTCLASS_SERVER,
+			LOG_DEBUG, name, "PMIx server initialized");
```

Comment:
no braces needed, only one statement in the else.

Reply:
This is also personal preference, but in this case I have seen bugs introduced when an additional line is added without adding braces. I'll remove the braces if you feel strongly.